daml/ContingentClaims/Financial.daml

```diff
@@ -20,51 +21,47 @@ unrollDates issueYear maturityYear fixingMonths fixingDay =
   date <$> [issueYear .. maturityYear] <*> fixingMonths <*> [fixingDay]
 
 -- | Forward agreement. Discounted by (potentially stochastic) interest rate `r`.
-forward : Observable f t x a => t -> f t x a -> Claim f t x a -> Claim f t x a
+forward : t -> Observation t x a -> Claim t x a -> Claim t x a
 forward maturity r payoff = When (at maturity) $ Scale r payoff
 
 -- | Forward rate agreement.
-fra : Observable f t x a => t -> t -> f t x a -> f t x a -> Claim f t x a -> Claim f t x a
+fra : t -> t -> Observation t x a -> Observation t x a -> Claim t x a -> Claim t x a
 fra t₁ t₂ r₀ r₁ = forward t₁ r₀ . forward t₂ r₁
 
 -- | Zero Coupon Bond.
-zcb : Observable f t x ccy => t -> x -> ccy -> Claim f t x ccy
+zcb : t -> x -> ccy -> Claim t x ccy
 zcb maturity principal ccy = forward maturity (O.pure principal) (One ccy)
 
--- | A floating rate bond. The first two arguments are `Observable`s.
-floating : Observable f t x ccy => f t x ccy -> f t x ccy -> ccy -> [t] -> Claim f t x ccy
-floating principal coupon asset = apo \case
-     [maturity] -> Left (forward maturity coupon (One asset)) `AndF`
-                   Left (forward maturity principal (One asset))
-     (t :: ts) -> Left (forward t coupon (One asset)) `AndF` Right ts
-     [] -> ZeroF
+-- | A floating rate bond.
+floating : forall t x ccy . Observation t x ccy -> Observation t x ccy -> ccy -> [t] -> Claim t x ccy
+floating principal coupon asset fixings = foldMap mkCoupon fixings <> mkPrincipal (last fixings : Optional t) -- TODO: not very efficient, but easy to understand
```

Comment:
@GeorgSchneider have a look at this - after adding a monoid instance i.e. `(<>)`, we can fold over claims quite nicely, like this implementation shows.